File: mo-system/mo-system-api/src/main/java/ink/ckx/mo/system/api/model/bo/RouteBO.java
package ink.ckx.mo.system.api.model.bo;

import ink.ckx.mo.system.api.enums.MenuTypeEnum;
import lombok.Data;

import java.util.List;

/**
 * 路由BO
 */
@Data
public class RouteBO {

    /**
     * 菜单id
     */
    private Long id;

    /**
     * 父菜单ID
     */
    private Long parentId;

    /**
     * 菜单名称
     */
    private String name;

    /**
     * 菜单类型(1-菜单；2-按钮)
     */
    private MenuTypeEnum type;

    /**
     * 路由路径(浏览器地址栏路径)
     */
    private String path;

    /**
     * 组件路径(vue页面完整路径，省略.vue后缀)
     */
    private String component;

    /**
     * 权限标识
     */
    private String perm;

    /**
     * 显示状态
     */
    private Integer visible;

    /**
     * 排序
     */
    private Integer sort;

    /**
     * 菜单图标
     */
    private String icon;

    /**
     * 如果设置为 true，目录没有子节点也会显示
     */
    private Boolean alwaysShow;

    /**
     * 拥有路由的权限
     */
    private List<String> roles;
}

File: mo-admin/mo-admin-api/src/main/kotlin/ink/ckx/mo/admin/api/model/bo/RouteBO.kt
package ink.ckx.mo.admin.api.model.bo

import ink.ckx.mo.admin.api.enums.MenuTypeEnum

/**
 * 路由BO
 */
data class RouteBO(

    /**
     * 菜单id
     */
    var id: Long? = null,

    /**
     * 父菜单ID
     */
    var parentId: Long? = null,

    /**
     * 菜单名称
     */
    var name: String? = null,

    /**
     * 菜单类型(1-目录；2-菜单；3-按钮)
     */
    var type: MenuTypeEnum? = null,

    /**
     * 路由路径(浏览器地址栏路径)
     */
    var path: String? = null,

    /**
     * 组件路径(vue页面完整路径，省略.vue后缀)
     */
    var component: String? = null,

    /**
     * 权限标识
     */
    var perm: String? = null,

    /**
     * 显示状态
     */
    var visible: Int? = null,

    /**
     * 排序
     */
    var sort: Int? = null,

    /**
     * 菜单图标
     */
    var icon: String? = null,

    /**
     * 如果设置为 true，目录没有子节点也会显示
     */
    var alwaysShow: Boolean? = null,

    /**
     * 拥有路由的权限
     */
    var roles: List<String>? = null,
)